### PR TITLE
AO-384 alerta update role

### PR DESCRIPTION
**Pull request by agelostsal**

### CODE REVIEW

File: roles/alerta/tasks/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    version: '{{alerta_ui_release}}'
+
+- name: Configure Alerta Dashboard
+  tags: alerta
+  template:
+    src: config.js.j2
+    dest: '{{ alerta_dashboard_dir }}/app/config.js'
+    mode: 0444
+
+# - name: Open http port
```
Comment (themiszamani) [inline roles/alerta/tasks/deploy.yml]: remove the comments.

File: roles/alerta/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+alerta_mail_debug: true
+
+# default: use pip to install argo-alert from github repo using devel branch
+argo_alert_repo: 'git+https://github.com/ARGOeu/argo-alert@devel#egg=argoalert'
+
+argo_alert_kafka: 'kafka.node.foo'
+argo_alert_token: 'secret_token'
+
+ca_bundle: true
+hostcert: /etc/grid-security/hostcert.pem
```
Comment (themiszamani) [inline roles/alerta/defaults/main.yml]: these are the firewall vars,
# firewall
cert_dir: /etc/grid-security/
cert_path: /etc/grid-security/hostcert.pem
cert_key_path: /etc/grid-security/hostkey.pem
ca_path: /etc/pki/tls/certs/
ca_path_bundle: /etc/pki/tls/certs/ca-bundle.crt

It is better to keep the same vars along all roles. Please update.

File: roles/alerta/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+hostkey: /etc/grid-security/hostkey.pem
+
+tenants:
```
Comment (themiszamani) [inline roles/alerta/defaults/main.yml]: tenants should change to alerta_tenants

File: roles/alerta/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+alerta_secret: ''
+alerta_token: ''
+alerta_server_dir: '/opt/alerta'
+alerta_wsgi_root_dir: '/var/www'
+alerta_dashboard_dir: '/var/www/alerta'
+alerta_dashboard_repo: https://github.com/alerta/angular-alerta-webui.git
+alerta_ui_release: 'v5.0.0'
+alerta_amqp_release: 'cb04fb6'
+alerta_mailer_release: 'e52f4f9'
+www_url: 'localhost'
```
Comment (themiszamani) [inline roles/alerta/defaults/main.yml]: update to alerta_

All alerta vars should follow the same rule

File: roles/alerta/molecule/default/Dockerfile.j2 [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Molecule managed
```
Comment (themiszamani) [inline roles/alerta/molecule/default/Dockerfile.j2]: remove it if we dont use it

File: roles/alerta/molecule/default/INSTALL.rst [hunk 6]
```diff
@@ -0,0 +1,26 @@
+*******
```
Comment (themiszamani) [inline roles/alerta/molecule/default/INSTALL.rst]: remove it if we dont use it

File: roles/alerta/molecule/default/molecule.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (themiszamani) [inline roles/alerta/molecule/default/molecule.yml]: update it based on what we use

File: roles/alerta/molecule/default/tests/test_default.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import os
```
Comment (themiszamani) [inline roles/alerta/molecule/default/tests/test_default.py]: remove it if we dont use it

File: roles/alerta/molecule/default/playbook.yml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (themiszamani) [inline roles/alerta/molecule/default/playbook.yml]: remove it if we dont use it

File: roles/alerta/templates/ssl.conf.j2 [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+#   considered compromised, too.
+#SSLCipherSuite RC4-SHA:AES128-SHA:HIGH:MEDIUM:!aNULL:!MD5
+#SSLHonorCipherOrder on
+
+#   Server Certificate:
+# Point SSLCertificateFile at a PEM encoded certificate.  If
+# the certificate is encrypted, then you will be prompted for a
+# pass phrase.  Note that a kill -HUP will prompt again.  A new
+# certificate can be generated using the genkey(1) command.
+SSLCertificateFile /etc/grid-security/hostcert.pem
```
Comment (themiszamani) [inline roles/alerta/templates/ssl.conf.j2]: use has_cert paths

cert_dir: /etc/grid-security/
cert_path: /etc/grid-security/hostcert.pem
cert_key_path: /etc/grid-security/hostkey.pem

File: roles/alerta/templates/ssl.conf.j2 [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+
+#   Server Private Key:
+#   If the key is not combined with the certificate, use this
+#   directive to point at the key file.  Keep in mind that if
+#   you've both a RSA and a DSA private key you can configure
+#   both in parallel (to also allow the use of DSA ciphers, etc.)
```
Comment (themiszamani) [inline roles/alerta/templates/ssl.conf.j2]: use has_cert paths

cert_dir: /etc/grid-security/
cert_path: /etc/grid-security/hostcert.pem
cert_key_path: /etc/grid-security/hostkey.pem

File: roles/alerta/templates/ssl.conf.j2 [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+SSLCertificateKeyFile /etc/grid-security/hostkey.pem
+
+#   Server Certificate Chain:
+#   Point SSLCertificateChainFile at a file containing the
+#   concatenation of PEM encoded CA certificates which form the
+#   certificate chain for the server certificate. Alternatively
+#   the referenced file can be the same as SSLCertificateFile
+#   when the CA certificates are directly appended to the server
+#   certificate for convinience.
+SSLCertificateChainFile /etc/grid-security/DigiCertCA.crt
```
Comment (themiszamani) [inline roles/alerta/templates/ssl.conf.j2]: use has_cert paths

cert_dir: /etc/grid-security/
cert_path: /etc/grid-security/hostcert.pem
cert_key_path: /etc/grid-security/hostkey.pem

File: roles/alerta/templates/ssmtp.conf.j2 [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+# Set this to never rewrite the "From:" line (unless not given) and to
+# use that address in the "from line" of the envelope.
+# FromLineOverride=YES
+
+# Use SSL/TLS to send secure messages to server.
+# UseTLS=YES
+#IMPORTANT: The following line is mandatory for TLS authentication
+# Use SSL/TLS certificate to authenticate against smtp host.
+# UseTLSCert=YES
+TLS_CA_File=/etc/pki/tls/certs/ca-bundle.crt
```
Comment (themiszamani) [inline roles/alerta/templates/ssmtp.conf.j2]: use ca_path_bundle: /etc/pki/tls/certs/ca-bundle.crt

File: roles/alerta/templates/ssmtp.conf.j2 [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+UseSTARTTLS=YES
+
+# Use this RSA certificate.
+TLSCert=/etc/grid-security/hostcert.pem
```
Comment (themiszamani) [inline roles/alerta/templates/ssmtp.conf.j2]: cert_path: /etc/grid-security/hostcert.pem
cert_key_path: /etc/grid-security/hostkey.pem

File: roles/alerta/templates/alerta-mailer.conf.j2 [hunk 15]
```diff
@@ -0,0 +1,27 @@
+[DEFAULT]
+timezone = Europe/Athens
+output = json
+endpoint = https://{{www_alias}}/api
+key = {{alerta_token}}
+sslverify = off
+timeout = 10.0
+debug = yes
```
Comment (kkoumantaros) [inline roles/alerta/templates/alerta-mailer.conf.j2]: debug should be no by default right ?

File: roles/alerta/molecule/default/INSTALL.rst [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+Install
+=======
+
+Ansible < 2.6
+
+.. code-block:: bash
+
+    $ sudo pip install docker-py
+
+Ansible >= 2.6
```
Comment (kkoumantaros) [inline roles/alerta/molecule/default/INSTALL.rst]: Should we consider as a given that ansible 2.6 is the one we use

**Pull request merged**